mlrun/pipelines/common/helpers.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import inspect
+from collections.abc import Iterator, MutableMapping
+from typing import Any, NoReturn
+
+project_annotation = "mlrun/project"
+run_annotation = "mlrun/pipeline-step-type"
+function_annotation = "mlrun/function-uri"
+
+
+class FlexibleMapper(MutableMapping):
```

Comment:
worth add docstring indication the incentive behind it

Reply:
for sure